lib/email_address/host.rb

```diff
@@ -352,8 +352,14 @@ def exchangers
     # Returns a DNS TXT Record
     def txt(alternate_host=nil)
       Resolv::DNS.open do |dns|
-        records = dns.getresources(alternate_host || self.dns_name,
+        dns.timeouts = @config[:dns_timeout] if @config[:dns_timeout]
```

Comment:
Cool! I didn't know about this feature. I didn't explore this much after reading ruby Timeout is [dangerous](https://jvns.ca/blog/2015/11/27/why-rubys-timeout-is-dangerous-and-thread-dot-raise-is-terrifying/). The docs says you can pass an array of retry times, so setting `dns_timeout: [3,3]` could be useful when one NS is down and you want more resiliency.